std/functional.d

```diff
@@ -260,6 +210,127 @@ unittest
     static assert(!is(typeof(binaryFun!FuncObj)));
 }
 
+/**
+Transforms a string representing an expression into a n-ary function.
+$(UL
+    $(LI The number of parameters is given by `argc`. If no explicit
+        `argc` is given, it is inferred to be `paramNames.length`.)
+    $(LI Explicit names for the function parameters may be given as
+        `paramNames`. If more than `argc` names are supplied, the
+        extras will be ignored.)
+    $(LI The letters (`a` through `z`) are used as default names
+        for up to the first 26 function parameters.)
+    $(LI `args[index]` can always be used to access any parameter by index,
+        regardless of whether it has been given a name.)
+)
+*/
+template stringLambda(alias fun, size_t argc, paramNames...)
+    if (paramNames.length == 0 || allSatisfy!(isSomeString, typeof(paramNames)))
+{
+    static if (is(typeof(fun) : string))
+    {
+        // No need to regenerate the aliases mixin for every ElementTypes permutation.
+        private enum aliases = stringLambdaAliases!(argc, paramNames);
```

Comment:
Since `stringLambdaAliases` is a template now, there is no longer any need to cache it at this level. (If you decide to leave it, update the comment - `ElementTypes` was renamed to `ArgTypes`.)